model/pdata/metrics.go

```diff
@@ -247,6 +247,22 @@ func (at AggregationTemporality) String() string {
 	return otlpmetrics.AggregationTemporality(at).String()
 }
 
+// DataPointFlags defines how a metric aggregator reports aggregated values.
+// It describes how those values relate to the time interval over which they are aggregated.
+type DataPointFlags uint32
+
+const (
+	// DataPointFlagsNone is the default DataPointFlags.
+	DataPointFlagsNone = DataPointFlags(otlpmetrics.DataPointFlags_FLAG_NONE)
+	// DataPointFlagsNoRecordedValue is a DataPointFlags for a metric aggregator which reports changes since last report time.
+	DataPointFlagsNoRecordedValue = DataPointFlags(otlpmetrics.DataPointFlags_FLAG_NO_RECORDED_VALUE)
```

Comment:
Question: Do we expect to do an `&` between these enums or to create all possible combinations? The current design may not be the best I think... I can see tho have `DataPointFlag` and have DataPointFlags(...DataPointFlag) or something like that.

What do you think?

Reply:
Any thoughts on the kind of scenario that would require this? Currently it seems like there wouldn't be a need for a None and a NoRecordedValue flag to be combined.

Reply:
It sounds like a `CombineDataPointFlags(...DataPointFlags) DataPointFlags` could be useful in the future, but as we have only one meaningful flag at this time it doesn't seem that we need it yet.

Reply:
That sounds good for merging, but for querying? You cannot do:
```
if point.Flags() == DataPointFlagsNoRecordedValue {
}
```

We need a way to say "Has" `DataPointFlag[s]NoRecordedValue` correct?

Reply:
So probably we need:
```
type DataPointFlags uint32

NewDataPointFlags(... DataPointFlag) {...}

func (dpf DataPointFlags) HasFlag(DataPointFlag) bool {}

type DataPointFlag uint32

const (
	// DataPointFlagNoRecordedValue is a DataPointFlag for a metric aggregator which reports changes since last report time.
	DataPointFlagNoRecordedValue = DataPointFlag(otlpmetrics.DataPointFlags_FLAG_NO_RECORDED_VALUE)
)

```

Something like this will allow us to add more flag values, because we force users to check for individual independent values in the flags.

Reply:
Ah I see. Will update the PR, thanks for clarifying 👍

Reply:
Added the functionality. Please take a look at resolve this conversation if it meets the requirements @bogdandrutu